spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/AbstractConnectionFactoryConfigurer.java

```diff
@@ -0,0 +1,61 @@
+package org.springframework.boot.autoconfigure.amqp;
+
+import org.springframework.amqp.rabbit.connection.AbstractConnectionFactory;
+import org.springframework.amqp.rabbit.connection.ConnectionNameStrategy;
+import org.springframework.boot.context.properties.PropertyMapper;
+import org.springframework.util.Assert;
+
+/**
+ * Configures {@link AbstractConnectionFactory Rabbit ConnectionFactory} with sensible
+ * defaults.
+ *
+ * @param <T> the connection factory type.
+ * @author Chris Bono
+ * @since 2.6.0
+ */
+public abstract class AbstractConnectionFactoryConfigurer<T extends AbstractConnectionFactory> {
+
+	private RabbitProperties rabbitProperties;
+
+	private ConnectionNameStrategy connectionNameStrategy;
+
+	public RabbitProperties getRabbitProperties() {
+		return rabbitProperties;
+	}
```

Comment:
As the properties are required, I'd that they're a constructor argument. I'll make that change as part of merging.